OrbitCore/Path.h

```diff
@@ -10,48 +10,47 @@
 
 class Path {
  public:
-  static void Init();
-
-  static std::string GetExecutableName();
-  static std::string GetExecutablePath();
-  static std::string GetBasePath();
-  static std::string GetDllPath(bool a_Is64Bit);
-  static std::string GetDllName(bool a_Is64Bit);
-  static std::string GetFileMappingFileName();
-  static std::string GetSymbolsFileName();
-  static std::string GetCachePath();
-  static std::string GetPresetPath();
-  static std::string GetPluginPath();
-  static std::string GetCapturePath();
-  static std::string GetDumpPath();
-  static std::string GetAppDataPath();
-  static std::string GetLogFilePath();
-  static std::string GetIconsPath();
-  static void Dump();
+  [[nodiscard]] static std::string GetExecutablePath();
+  [[nodiscard]] static std::string GetExecutableDir();
+  [[nodiscard]] static std::string GetBasePath();
+  [[nodiscard]] static std::string GetDllPath(bool a_Is64Bit);
+  [[nodiscard]] static std::string GetDllName(bool a_Is64Bit);
+  [[nodiscard]] static std::string GetFileMappingFileName();
+  [[nodiscard]] static std::string GetSymbolsFileName();
+  [[nodiscard]] static std::string CreateAndGetCacheDir();
+  [[nodiscard]] static std::string CreateAndGetPresetDir();
+  [[nodiscard]] static std::string CreateAndGetPluginDir();
+  [[nodiscard]] static std::string CreateAndGetCaptureDir();
+  [[nodiscard]] static std::string CreateAndGetDumpDir();
+  [[nodiscard]] static std::string CreateAndGetOrbitAppDataDir();
```

Comment:
Since the create operation will happen only once, should we rename to CreateOrGetDir instead of CreateANDGet? We use the GetOrCreateX pattern elsewhere in the code, GetOrCreateThreadTrack for example, maybe we should do the same here for uniformity.

Reply:
True, I changed this to be more consistent